### PR TITLE
Functional tests - Refacto 'Sort, pagination and bulk delete features test'

### PR DESCRIPTION
**Pull request by nesrineabdmouleh**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Refacto 'Sort, pagination and bulk delete features test'
| Type?             | refacto
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | no need
| Fixed ticket?     | no
| Related PRs       | no
| Sponsor company   | no

### CODE REVIEW

**Review by boubkerbribri — APPROVED** [review]

**Review by boubkerbribri — CHANGES_REQUESTED** [review]
Some conflicts to resolve here

File: tests/UI/commonTests/BO/catalog/features.ts [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+      const pageTitle: string = await attributesPage.getPageTitle(page);
+      await expect(pageTitle).to.contains(attributesPage.pageTitle);
+    });
+
+    it('should go to Features page', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'goToFeaturesPage', baseContext);
+
+      await attributesPage.goToFeaturesPage(page);
+
+      const pageTitle: string = await featuresPage.getPageTitle(page);
```
Comment (Progi1984) [inline tests/UI/commonTests/BO/catalog/features.ts]:
```suggestion
      const pageTitle = await featuresPage.getPageTitle(page);
```

File: tests/UI/commonTests/BO/catalog/features.ts [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+      numberOfFeatures = await featuresPage.resetAndGetNumberOfLines(page);
+      await expect(numberOfFeatures).to.be.above(0);
+    });
+
+    it('should go to add new feature page', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'goToAddNewFeaturePage', baseContext);
+
+      await featuresPage.goToAddFeaturePage(page);
+
+      const pageTitle: string = await addFeaturePage.getPageTitle(page);
```
Comment (Progi1984) [inline tests/UI/commonTests/BO/catalog/features.ts]:
```suggestion
      const pageTitle = await addFeaturePage.getPageTitle(page);
```

File: tests/UI/commonTests/BO/catalog/features.ts [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+      await expect(pageTitle).to.contains(addFeaturePage.createPageTitle);
+    });
+
+    it('should create feature', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'createNewFeature', baseContext);
+
+      const textResult: string = await addFeaturePage.setFeature(page, createFeatureData);
```
Comment (Progi1984) [inline tests/UI/commonTests/BO/catalog/features.ts]:
```suggestion
      const textResult = await addFeaturePage.setFeature(page, createFeatureData);
```

File: tests/UI/commonTests/BO/catalog/features.ts [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+      await testContext.addContextItem(this, 'testIdentifier', 'goToAttributesPage', baseContext);
+
+      await dashboardPage.goToSubMenu(
+        page,
+        dashboardPage.catalogParentLink,
+        dashboardPage.attributesAndFeaturesLink,
+      );
+      await attributesPage.closeSfToolBar(page);
+
+      const pageTitle: string = await attributesPage.getPageTitle(page);
```
Comment (Progi1984) [inline tests/UI/commonTests/BO/catalog/features.ts]:
```suggestion
      const pageTitle = await attributesPage.getPageTitle(page);
```

File: tests/UI/commonTests/BO/catalog/features.ts [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+      await expect(pageTitle).to.contains(attributesPage.pageTitle);
+    });
+
+    it('should go to Features page', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'goToFeaturesPage', baseContext);
+
+      await attributesPage.goToFeaturesPage(page);
+
+      const pageTitle: string = await featuresPage.getPageTitle(page);
```
Comment (Progi1984) [inline tests/UI/commonTests/BO/catalog/features.ts]:
```suggestion
      const pageTitle = await featuresPage.getPageTitle(page);
```

File: tests/UI/commonTests/BO/catalog/features.ts [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+      numberOfFeatures = await featuresPage.resetAndGetNumberOfLines(page);
+      await expect(numberOfFeatures).to.be.above(0);
+    });
+
+    it(`should filter by feature name '${featureName}'`, async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'filterToBulkDelete', baseContext);
+
+      await featuresPage.filterTable(page, 'b!name', featureName);
+
+      const numberOfFeaturesAfterFilter: number = await featuresPage.getNumberOfElementInGrid(page);
```
Comment (Progi1984) [inline tests/UI/commonTests/BO/catalog/features.ts]:
```suggestion
      const numberOfFeaturesAfterFilter = await featuresPage.getNumberOfElementInGrid(page);
```

File: tests/UI/commonTests/BO/catalog/features.ts [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+      await testContext.addContextItem(this, 'testIdentifier', 'getNumberToDelete', baseContext);
+
+      numberOfFeaturesToDelete = await featuresPage.getNumberOfElementInGrid(page);
+      await expect(numberOfFeaturesToDelete).to.be.above(0);
+    });
+
+    it('should delete features by Bulk Actions and check result', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'bulkDeleteFeatures', baseContext);
+
+      const deleteTextResult: string = await featuresPage.bulkDeleteFeatures(page);
```
Comment (Progi1984) [inline tests/UI/commonTests/BO/catalog/features.ts]:
```suggestion
      const deleteTextResult = await featuresPage.bulkDeleteFeatures(page);
```

File: tests/UI/commonTests/BO/catalog/features.ts [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+      await expect(deleteTextResult).to.be.contains(featuresPage.successfulMultiDeleteMessage);
+    });
+
+    it('should reset all filters', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'resetFilter', baseContext);
+
+      const numberOfFeaturesAfterDelete: number = await featuresPage.resetAndGetNumberOfLines(page);
```
Comment (Progi1984) [inline tests/UI/commonTests/BO/catalog/features.ts]:
```suggestion
      const numberOfFeaturesAfterDelete = await featuresPage.resetAndGetNumberOfLines(page);
```

**Review by Progi1984 — COMMENTED** [review]
Some feedbacks

File: tests/UI/commonTests/BO/catalog/features.ts (hunk 8 above)
Comment (nesrineabdmouleh) [reply tests/UI/commonTests/BO/catalog/features.ts]: Done thx :+1:

**Review by Progi1984 — APPROVED** [review]

**Review by boubkerbribri — APPROVED** [review]

**Comment by boubkerbribri** [comment]
Thanks @nesrineabdmouleh

**Pull request merged**